code/modules/projectiles/guns/projectile/launchers.dm

```diff
@@ -83,3 +83,752 @@
+	var/datum/effect_system/explosion/smoke/E = new/datum/effect_system/explosion/smoke()
+	E.set_up(target)
+	E.start()
+	return TRUE
+
+/obj/item/projectile/a66mm/on_range()
+	on_hit(get_turf(src))
+	..()
+
+/obj/item/projectile/a66mm/rod
+	name ="66mm Rod rocket"
+	desc = "A rocket designed to punch through almost anything."
+	icon_state = "SLRW-rocket"
+	alwayslog = FALSE
+	rocket_stop = null
+	ex_range = 1
+
+/obj/item/projectile/a66mm/rod/Initialize()
+	..()
+	rocket_stop -= /turf/simulated/wall
+	rocket_stop -= /obj/structure/girder
+	rocket_stop -= /obj/machinery/door/airlock/highsecurity
+	rocket_stop += /turf/simulated/wall/r_wall
+
+/obj/item/projectile/a66mm/rod/prehit(turf/target as turf)
+	if(!..())
+		return FALSE
+	if(istype(target, /turf/simulated/wall))
+		sfx_hit(60, 40)
+		target.ChangeTurf(/turf/simulated/floor/plating)
+		ffx_hit(target)
+	return TRUE
+
+/obj/item/projectile/a66mm/rod/on_hit(turf/target, blocked = 0)
+	if(locate(/obj/mecha) in target.contents)
+		for(var/obj/mecha/M in target.contents)
+			if(!is_fellow_antag(M) && M.density)
+				M.take_damage(damage, BRUTE)
+				break
+	else if(locate(/mob/living) in target.contents)
+		for(var/mob/living/L in target.contents)
+			if(!is_fellow_antag(L) && L.density)
+				L.apply_damage(damage, BRUTE)
+				L.check_projectile_dismemberment(src, "chest")
+				break
+
+	if(friendly_fire_dud(target, ex_range))
+		return FALSE
+
+	for(var/turf/T in range(ex_range, target))
+		ffx_hit(T)
+
+	explosion(target, 1, -1, ex_range+1, flame_range = ex_range, cause = "[type] fired by [key_name(firer)]")
+	var/datum/effect_system/explosion/smoke/E = new/datum/effect_system/explosion/smoke()
+	E.set_up(target)
+	E.start()
+	return TRUE
+
+/obj/item/projectile/a66mm/frag
+	name ="66mm Frag rocket"
+	desc = "A rocket designed to maim soft targets in a wide area."
+	icon_state = "SLRW-rocket"
+	ex_range = 3
+
+/obj/item/projectile/a66mm/frag/on_hit(turf/target)
+	if(locate(/mob/living) in target.contents)
+		for(var/mob/living/L in target.contents)
+			if(!is_fellow_antag(L) && L.density)
+				L.apply_damage(damage, BRUTE)
+				L.check_projectile_dismemberment(src, "chest")
+				break
+	else
+		target = get_turf(src)
+
+	if(friendly_fire_dud(target, ex_range))
+		return FALSE
+
+	for(var/mob/living/M in view(ex_range, target))
+		if(istype(M, /mob/living/carbon))
+			var/mob/living/carbon/C = M
+			if(!C.check_eye_prot())
+				C.EyeBlurry(10)
+			if(C.check_ear_prot())
+				C.AdjustEarDamage(5, 20)
+			var/obj/item/embedded/S = new /obj/item/embedded/shrapnel
+			C.hitby(S, skipcatch = TRUE)
+			S.throwforce = 1
+			S.throw_speed = 1
+			C.bleed(25)
+			to_chat(M, "<span class='warning'>You're hit by shrapnel!</span>")
+		else if(istype(M, /mob/living/silicon))
+			to_chat(M, "<span class='warning'>Shrapnel bounces off you!</span>")
+		else
+			to_chat(M, "<span class='warning'>Shrapnel misses you!</span>")
+
+	exsfx_hit(ex_range+4, ex_range+10, 'sound/effects/explosion1.ogg')
```

Comment:
spacing

Reply:
Could you explain what spacing needs to happen?

Reply:
any spacing between operators such as `+` or `-` etc. Also applies to commas and such

Reply:
so like...

`ex_range + 4 , ex_range + 10 ,`
or
`ex_range+4,ex_range+10,`

Reply:
The upper one